Zend/zend_compile.c

```diff
@@ -8462,6 +8462,15 @@ static zend_op_array *zend_compile_func_decl_ex(
 	/* Pop the loop variable stack separator */
 	zend_stack_del_top(&CG(loop_var_stack));
 
+	if (op_array->fn_flags & ZEND_ACC_CLOSURE) {
```

Comment:
In https://github.com/php/php-src/issues/10782#issuecomment-1457177732 there is also a condition to skip this for traits, not sure, but highly likely there is a missing test for it with observer.